### DISCUSSION TITLE
Is there a way to get Renovate to only prune orphaned branches without active PRs, and not autoclose PRs?

### DISCUSSION THREAD

**Discussion by ajburley** [renovatebot/renovate#31881] [Request Help] [answered]
### What would you like help with?

I would like help with my configuration

### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.

Azure DevOps

### Please tell us more about your question or problem

We didn't want Renovate to autoclose PRs as sometimes we might be in the middle of working on a PR which Renovate originally created (e.g. committing some additional changes to that branch before merging). So we currently have set `pruneStaleBranches=false`. However we also have a rule in our organization that when a branch was authored by someone else, it can't be deleted except by an admin. This means that the branches which Renovate created and then we merged/closed those PRs, they just stay there until we can get them deleted by an admin. I'm wondering if there's a way to get Renovate to close its own branches if they are not part of an active PR, and not autoclose PRs?

### Logs (if relevant)

<details><summary>Logs</summary>

```

Replace this text with your logs, between the starting and ending triple backticks

```

</details>

**Comment by rarkins** [comment]
No, it's not possible. But I'm curious about what circumstance leads to:
- A PR which Renovate believes is no longer necessary and autocloses, but
- You believe it's necessary and are "working on" it

Further, Renovate won't autoclose a PR if you have already committed to it, so it must be an edge case where:
- You start "working on" the PR but haven't pushed a commit yet
- Renovate determines it's no longer needed before you have a chance to commit

Just how often did that happen that it lead you to disable pruning?

**Comment by ajburley** [comment]
Hi @rarkins thanks for your reply. It was not based on experience with Renovate but just based on reading the documentation and available other info such as https://github.com/renovatebot/renovate/issues/2848 and coming to the conclusion that it could cause issues for us. We didn't realise that Renovate won't autoclose a PR if we have already committed to it (is that documented anywhere?) Based on that, we can probably enable pruning again.

**Reply by rarkins** [reply, marked as answer]
https://github.com/renovatebot/renovate/pull/31894/files

**Comment by ajburley** [comment]
Thank you very much, that's really useful info.
